src/api/java/com/minecolonies/api/colony/fields/AbstractFieldView.java

```diff
@@ -0,0 +1,132 @@
+package com.minecolonies.api.colony.fields;
+
+import com.minecolonies.api.colony.IColonyView;
+import com.minecolonies.api.colony.buildings.views.IBuildingView;
+import com.minecolonies.api.colony.fields.registry.FieldRegistries;
+import com.minecolonies.api.util.BlockPosUtil;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.FriendlyByteBuf;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Abstract implementation of the field it's view instance.
+ */
+public abstract class AbstractFieldView implements IFieldView
+{
+    /**
+     * Colony owning the field.
+     */
+    private final @NotNull IColonyView colony;
+
+    /**
+     * The position of the field.
+     */
+    private final @NotNull BlockPos position;
+
+    /**
+     * The type of the field.
+     */
+    private FieldRegistries.FieldEntry fieldType = null;
+
+    /**
+     * Building id of the building owning the field.
+     */
+    private BlockPos buildingId;
+
+    /**
+     * Constructor used in deserialization.
+     *
+     * @param colony the colony this field belongs to.
+     */
+    protected AbstractFieldView(@NotNull IColonyView colony, @NotNull BlockPos position)
```

Comment:
Nothing of these values cannot be sent to the client side, we can reuse the same field class for both sides and make it IColony